packages/rtn-web-browser/ios/AmplifyRTNWebBrowser.swift

```diff
@@ -0,0 +1,63 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import Foundation
+import AuthenticationServices
+
+private class PresentationContextProvider: NSObject, ASWebAuthenticationPresentationContextProviding {
+    func presentationAnchor(for session: ASWebAuthenticationSession) -> ASPresentationAnchor {
+        return ASPresentationAnchor()
+    }
+}
+
+@objc(AmplifyRTNWebBrowser)
+class AmplifyRTNWebBrowser: NSObject {
+    var webBrowserAuthSession: ASWebAuthenticationSession?
+    private let presentationContextProvider = PresentationContextProvider()
+    
+    private func isUrlValid(url: URL) -> Bool {
+        return url.scheme == "http" || url.scheme == "https"
```

Comment:
nit: maybe using caseInsensitiveCompare
```suggestion
     url.scheme != nil && 
     ["http", "https"].contains { $0.caseInsensitiveCompare(url.scheme!) == .orderedSame } 
```